public/app/plugins/datasource/prometheus/querybuilder/shared/QueryOptionGroup.tsx

```diff
@@ -10,16 +10,28 @@ import { QueryStats } from 'app/plugins/datasource/loki/types';
 export interface Props {
   title: string;
   collapsedInfo: string[];
-  queryStats?: QueryStats | null;
+  queryStats?: QueryStats | string | null;
   children: React.ReactNode;
 }
 
 export function QueryOptionGroup({ title, children, collapsedInfo, queryStats }: Props) {
   const [isOpen, toggleOpen] = useToggle(false);
   const styles = useStyles2(getStyles);
 
+  const generateQueryStats = () => {
```

Comment:
i recommend moving `generateQueryStats` and `convertUnits` outside of `QueryOptionsGroup`. they only use the `queryStats` value, so i think it's easier to think about them when they are simple functions that take the `queryStats` and return something (will be a bit faster too).